src/annotator/config/test/index-test.js

```diff
@@ -51,24 +67,117 @@ describe('annotator/config/index', function () {
     });
   });
 
-  ['appType', 'openSidebar', 'subFrameIdentifier'].forEach(function (
-    settingName
-  ) {
-    it(`reads ${settingName} from the host page, even when in a browser extension`, function () {
-      getConfig('all', 'WINDOW');
-      assert.calledWith(
-        fakeSettingsFrom().hostPageSetting,
-        settingName,
-        sinon.match({ allowInBrowserExt: true })
-      );
+  describe('browser extension', () => {
+    context('browser extension set to true', function () {
+      it('returns only config values where `allowInBrowserExt` is true or the defaultValue if provided', () => {
+        fakeIsBrowserExtension.returns(true);
+        const config = getConfig('all', 'WINDOW');
+        assert.deepEqual(
+          {
+            appType: 'fakeValue',
+            annotations: 'fakeValue',
+            branding: null,
+            clientUrl: 'fakeValue',
+            enableExperimentalNewNoteButton: null,
+            externalContainerSelector: null,
+            focus: null,
+            group: null,
+            notebookAppUrl: 'fakeValue',
+            onLayoutChange: null,
+            openSidebar: true, // coerced
+            query: 'fakeValue',
+            requestConfigFromFrame: null,
+            services: null,
+            showHighlights: null,
+            sidebarAppUrl: 'fakeValue',
+            subFrameIdentifier: 'fakeValue',
+            theme: null,
+            usernameUrl: null,
+          },
+          config
+        );
+      });
+    });
+
+    context('browser extension set to false', function () {
```

Comment:
```suggestion
    context('when client is not loaded from browser extension', () => {
```

See previous comments.